### PR TITLE
Header and Util changes

### PR DESCRIPTION
**Pull request by Lethalinjectionx**
Changes:

1. The `AddItem` method in the `UPrimalInventoryComponent` struct in `Inventory.h` has been modified. The parameter `theItemInfo` has been changed from a pointer to a reference. This change enhances the safety and readability of the code by ensuring that the parameter cannot be null and that it cannot be changed to point to a different object. (See `Inventory.h`)

2. The `ServerEquipItem` method in the `UPrimalInventoryComponent` struct in `Inventory.h` has been modified. The parameter `itemID` has been changed from a pointer to a reference. This change enhances the safety and readability of the code by ensuring that the parameter cannot be null and that it cannot be changed to point to a different object. (See `Inventory.h`)

3. The `GetBlueprint` method in the `AsaApi` namespace in `ArkApiUtils.h` has been modified. The method now replaces "Default__" with an empty string in the path name of the blueprint. If the object or its class is null, the method now explicitly returns an empty string. This change makes the method's behavior more predictable and its output more useful. (See `ArkApiUtils.h`)

### CODE REVIEW

File: AsaApi/Core/Public/API/ARK/Inventory.h [hunk 1]
```diff
@@ -960,15 +960,15 @@ struct UPrimalInventoryComponent : UActorComponent
...
+    UPrimalItem* AddItem(const FItemNetInfo* theItemInfo, bool bEquipItem, bool bAddToSlot, bool bDontStack, FItemNetID* InventoryInsertAfterItemID, bool bShowHUDNotification, bool bDontRecalcSpoilingTime, bool bForceIncompleteStacking, AShooterCharacter* OwnerPlayer, bool bIgnoreAbsoluteMaxInventory, bool bInsertAtItemIDIndexInstead, bool doVersionCheck) { return NativeCall<UPrimalItem*, const FItemNetInfo*, bool, bool, bool, FItemNetID*, bool, bool, bool, AShooterCharacter*, bool, bool, bool>(this, "UPrimalInventoryComponent.AddItem(FItemNetInfo&,bool,bool,bool,FItemNetID*,bool,bool,bool,AShooterCharacter*,bool,bool,bool)", theItemInfo, bEquipItem, bAddToSlot, bDontStack, InventoryInsertAfterItemID, bShowHUDNotification, bDontRecalcSpoilingTime, bForceIncompleteStacking, OwnerPlayer, bIgnoreAbsoluteMaxInventory, bInsertAtItemIDIndexInstead, doVersionCheck); }
```
Comment (SubstituteR) [inline AsaApi/Core/Public/API/ARK/Inventory.h:963]: Native calls is using references in the function names, but the function signatures are still taking pointers.

File: AsaApi/Core/Public/API/ARK/Inventory.h [hunk 2]
```diff
@@ -960,15 +960,15 @@ struct UPrimalInventoryComponent : UActorComponent
...
     ADroppedItem* EjectItem(const FItemNetID* itemID, bool bPreventImpule, bool bForceEject, bool bSetItemLocation, const UE::Math::TVector<double>* LocationOverride, bool showHUDMessage, TSubclassOf<ADroppedItem> TheDroppedTemplateOverride, bool bAssignToTribeForPickup, int AssignedTribeID) { return NativeCall<ADroppedItem*, const FItemNetID*, bool, bool, bool, const UE::Math::TVector<double>*, bool, TSubclassOf<ADroppedItem>, bool, int>(this, "UPrimalInventoryComponent.EjectItem(FItemNetID*,bool,bool,bool,UE::Math::TVector<double>*,bool,TSubclassOf<ADroppedItem>,bool,int)", itemID, bPreventImpule, bForceEject, bSetItemLocation, LocationOverride, showHUDMessage, TheDroppedTemplateOverride, bAssignToTribeForPickup, AssignedTribeID); }
-    bool ServerEquipItem(FItemNetID* itemID) { return NativeCall<bool, FItemNetID*>(this, "UPrimalInventoryComponent.ServerEquipItem(FItemNetID*)", itemID); }
+    bool ServerEquipItem(FItemNetID* itemID) { return NativeCall<bool, FItemNetID*>(this, "UPrimalInventoryComponent.ServerEquipItem(FItemNetID&)", itemID); }
```
Comment (SubstituteR) [inline AsaApi/Core/Public/API/ARK/Inventory.h:971]: Here also

File: AsaApi/Core/Public/Ark/ArkApiUtils.h [hunk 3]
```diff
@@ -604,10 +604,11 @@ namespace AsaApi
 		{
 			if (object != nullptr && object->ClassPrivateField() != nullptr)
 			{
-				return GetClassBlueprint(object->ClassPrivateField());
+				FString path_name = GetClassBlueprint(object->ClassPrivateField());
+				return path_name.Replace(L"Default__", L"", ESearchCase::CaseSensitive);
 			}
-
-			return FString("");
+			else
```
Comment (SubstituteR) [inline AsaApi/Core/Public/Ark/ArkApiUtils.h:610]: Why if/else and not if/default?

**Review by SubstituteR — CHANGES_REQUESTED** [review]

File: AsaApi/Core/Public/API/ARK/Inventory.h (hunk 1 above)
Comment (Lethalinjectionx) [reply AsaApi/Core/Public/API/ARK/Inventory.h]: pushed a commit to correct

File: AsaApi/Core/Public/API/ARK/Inventory.h (hunk 2 above)
Comment (Lethalinjectionx) [reply AsaApi/Core/Public/API/ARK/Inventory.h]: pushed a commit to correct

File: AsaApi/Core/Public/Ark/ArkApiUtils.h (hunk 3 above)
Comment (Lethalinjectionx) [reply AsaApi/Core/Public/Ark/ArkApiUtils.h]: pushed a commit to correct

**Pull request merged**